utilities/virt.py

```diff
@@ -2387,3 +2388,144 @@ def validate_libvirt_persistent_domain(vm):
 def get_nodes_gpu_info(util_pods, node):
     pod_exec = utilities.infra.ExecCommandOnPod(utility_pods=util_pods, node=node)
     return pod_exec.exec(command="sudo /sbin/lspci -nnk | grep -A 3 '3D controller'")
+
+
+def assert_linux_efi(vm):
+    """
+    Verify guest OS is using EFI.
+    """
+    run_ssh_commands(host=vm.ssh_exec, commands=["ls", "-ld", "/sys/firmware/efi"])
+
+
+def pause_optional_migrate_unpause_and_check_connectivity(vm, migrate=False):
+    vmi = VirtualMachineInstance(client=get_client(), name=vm.vmi.name, namespace=vm.vmi.namespace)
+    vmi.pause(wait=True)
+    if migrate:
+        migrate_vm_and_verify(vm=vm, wait_for_interfaces=False, check_ssh_connectivity=False)
+    vmi.unpause(wait=True)
+    LOGGER.info("Verify VM is running and ready after unpause")
+    wait_for_running_vm(vm=vm)
+
+
+def validate_pause_optional_migrate_unpause_linux_vm(vm, pre_pause_pid=None, migrate=False):
+    proc_name = OS_PROC_NAME["linux"]
+    if not pre_pause_pid:
+        pre_pause_pid = start_and_fetch_processid_on_linux_vm(vm=vm, process_name=proc_name, args="localhost")
+    pause_optional_migrate_unpause_and_check_connectivity(vm=vm, migrate=migrate)
+    post_pause_pid = fetch_pid_from_linux_vm(vm=vm, process_name=proc_name)
+    kill_processes_by_name_linux(vm=vm, process_name=proc_name)
+    assert post_pause_pid == pre_pause_pid, (
+        f"PID mismatch!\nPre pause PID is: {pre_pause_pid}\nPost pause PID is: {post_pause_pid}"
```

Comment:
just a nit - for better readability:
```suggestion
        f"PID mismatch!\n Pre pause PID is: {pre_pause_pid}\n Post pause PID is: {post_pause_pid}"
```